nix-tools/make-install-plan/MakeInstallPlan.hs

```diff
@@ -0,0 +1,128 @@
+          cliConfig = commandLineFlagsToProjectConfig globalFlags flags mempty
+       in installPlanAction verbosity cliConfig
+
+cmdUI :: CommandUI (NixStyleFlags ())
+cmdUI =
+  CommandUI
+    { commandName = "",
+      commandSynopsis = "Makes an install-plan",
+      commandUsage = ("Usage: " ++),
+      commandDescription = Nothing,
+      commandNotes = Nothing,
+      commandDefaultFlags = defaultNixStyleFlags (),
+      commandOptions = nixStyleOptions (const [])
+    }
+
+installPlanAction :: Verbosity -> ProjectConfig -> IO ()
+installPlanAction verbosity cliConfig = do
+  ProjectBaseContext {distDirLayout, cabalDirLayout, projectConfig, localPackages} <-
+    establishProjectBaseContext verbosity cliConfig OtherCommand
+
+  -- Two variants of the install plan are returned: with and without
+  -- packages from the store. That is, the "improved" plan where source
+  -- packages are replaced by pre-existing installed packages from the
+  -- store (when their ids match), and also the original elaborated plan
+  -- which uses primarily source packages.
+  (_improvedPlan, elaboratedPlan, elaboratedSharedConfig, _tis, _at) <-
+    rebuildInstallPlan verbosity distDirLayout cabalDirLayout projectConfig localPackages
+
+  -- Write plan.json
+  Cabal.notice verbosity $ "Writing plan.json to " ++ distProjectCacheFile distDirLayout "plan.json"
+  writePlanExternalRepresentation distDirLayout elaboratedPlan elaboratedSharedConfig
+
+  -- Write cabal.freeze
+  let cabalFreezeFile = distProjectFile distDirLayout "freeze"
+  Cabal.notice verbosity $ "Wrote freeze file to " ++ cabalFreezeFile
+  writeProjectConfigFile cabalFreezeFile projectConfig
+
+  -- Write cabal files and their nix version
+  let cabalFilesDir = distDirectory distDirLayout </> "cabal-files"
+  Cabal.createDirectoryIfMissingVerbose verbosity True cabalFilesDir
+  Cabal.notice verbosity $ "Writing cabal files to " ++ cabalFilesDir
+
+  let ecps = [ecp | InstallPlan.Configured ecp <- InstallPlan.toList elaboratedPlan, not $ elabLocalToProject ecp]
+
+  for_ ecps $
+    \ElaboratedConfiguredPackage
+       { elabPkgSourceId,
+         elabPkgSourceLocation,
+         elabPkgSourceHash,
+         elabPkgDescriptionOverride
+       } -> do
+        let pkgFile = cabalFilesDir </> prettyShow (pkgName elabPkgSourceId) <.> "cabal"
+        let nixFile = cabalFilesDir </> prettyShow (pkgName elabPkgSourceId) <.> "nix"
+        for_ elabPkgDescriptionOverride $ \pkgTxt ->
+          do
+            -- raw
+            Cabal.info verbosity $ "Writing cabal file for " ++ prettyShow elabPkgSourceId ++ " to " ++ pkgFile
+            BSL.writeFile pkgFile pkgTxt
+            -- nix
+            let src = packageLocation2Src elabPkgSourceLocation elabPkgSourceHash
+            Cabal2Nix.cabal2nix False Cabal2Nix.MinimalDetails (Just src) (Cabal2Nix.OnDisk pkgFile)
+            >>= writeDoc nixFile . prettyNix
+
+packageLocation2Src :: PackageLocation local -> Maybe HashValue -> Cabal2Nix.Src
```

Comment:
Should this helper function be in `Cabal2Nix` maybe? I'm surprised something like this isn't used somewhere else already.